content/en/docs/Writing policies/external-data-sources.md

```diff
@@ -0,0 +1,440 @@
+
+The output of this will be `apps/v1`. Older versions of Kubernetes (prior to 1.18) will also show `apps/v1beta2`.
+
+{{% /alert %}}
+
+
+### Handling collections
+
+The API server response for a `HTTP GET` on a URL path that requests collections of resources will be an object with a list of items (resources).
+
+Here is an example that fetches all namespaces:
+
+```
+kubectl get --raw /api/v1/namespaces | jq
+```
+
+{{% alert title="Tip" color="info" %}}
+Use [`jq`](https://stedolan.github.io/jq/) to format output for readability.  
+{{% /alert %}}
+
+
+This will return a `NamespaceList` object with a property `items` that contains the list of namespaces:
+
+```json
+{
+    "kind": "NamespaceList",
+    "apiVersion": "v1",
+    "metadata": {
+      "selfLink": "/api/v1/namespaces",
+      "resourceVersion": "2009258"
+    },
+    "items": [
+      {
+        "metadata": {
+          "name": "default",
+          "selfLink": "/api/v1/namespaces/default",
+          "uid": "5011b5d5-abb7-4fef-93f9-8b5fa4b2eba9",
+          "resourceVersion": "155",
+          "creationTimestamp": "2021-01-19T20:20:37Z",
+          "managedFields": [
+            {
+              "manager": "kube-apiserver",
+              "operation": "Update",
+              "apiVersion": "v1",
+              "time": "2021-01-19T20:20:37Z",
+              "fieldsType": "FieldsV1",
+              "fieldsV1": {
+                "f:status": {
+                  "f:phase": {}
+                }
+              }
+            }
+          ]
+        },
+        "spec": {
+          "finalizers": [
+            "kubernetes"
+          ]
+        },
+        "status": {
+          "phase": "Active"
+        }
+      },
+
+      ...
+
+```
+
+To process this data in JMESPath, reference the `items`. Here is an example, to extract a few metadata fields across all namespaces:
+
+```sh
+kubectl get --raw /api/v1/namespaces | jp "items[*].{name: metadata.name, creationTime: metadata.creationTimestamp}"
+```
+
+This produces a new JSON list of objects with properties `name` and `creationTime`. 
+
+```json
+[
+  {
+    "creationTimestamp": "2021-01-19T20:20:37Z",
+    "name": "default"
+  },
+  {
+    "creationTimestamp": "2021-01-19T20:20:36Z",
+    "name": "kube-node-lease"
+  },
+
+  ...
+
+```
+
+To find an item in the list you can use JMESPath filters. For example, this command will match a namespace by its name:
+
+```sh
+ kubectl get --raw /api/v1/namespaces | jp "items[?metadata.name == 'default'].{uid: metadata.uid, creationTimestamp: metadata.creationTimestamp}"
+```
+
+In addition to wildcards and filters, JMESPath has many additional powerful features including several useful functions. Be sure to go through the [JMESPath tutorial](https://jmespath.org/tutorial.html) and try the interactive examples.
+
+
+### Sample Policy: Limit Services of type LoadBalancer in a namespace 
+
+Here is a complete sample policy that limits each namespace to a single service of type `LoadBalancer`. 
```

Comment:
Can you provide just a short list of the filters that are mostly likely to be used by Kyverno users? For example, "length" would be a common one. Most of them I don't find to be likely used in Kyverno.

Reply:
Yes, same. Will need to add when we support custom JMESPath functions.

Reply:
I probably wouldn't provide an exhaustive list. Users can follow the links you provide if they want that. A curated list of those which are most valuable seems to be better.